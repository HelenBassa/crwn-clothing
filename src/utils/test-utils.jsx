import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };