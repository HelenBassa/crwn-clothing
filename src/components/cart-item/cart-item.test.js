import { render } from "../../utils/test-utils";
import CartItem from "./cart-item.component";

test("should render CartItem component", () => {
  const mockItem = {
    name: "Adidas NMD",
    imageUrl: "https://i.ibb.co/0s3pdnc/adidas-nmd.png",
    price: 220,
    quantity: 2,
  };

  const { container } = render(<CartItem cartItem={mockItem} />);
  expect(container).toMatchSnapshot();
});
