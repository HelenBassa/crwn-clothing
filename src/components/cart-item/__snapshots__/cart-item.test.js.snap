// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartItem component 1`] = `
<div>
  <div
    class="sc-bcXHqe kasWdQ"
  >
    <img
      alt="Adidas NMD"
      src="https://i.ibb.co/0s3pdnc/adidas-nmd.png"
    />
    <div
      class="sc-gswNZR gMScrp"
    >
      <span>
        Adidas NMD
      </span>
      <span>
        2
         x $
        220
      </span>
    </div>
  </div>
</div>
`;
